modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/portlet/action/EditCommerceOrderMVCRenderCommand.java

```diff
@@ -59,6 +62,26 @@ public String render(
 				commerceOrderContentDisplayContext.getCommerceOrder();
 
 			if ((commerceOrder != null) && commerceOrder.isOpen()) {
+				ThemeDisplay themeDisplay =
+					(ThemeDisplay)renderRequest.getAttribute(
+						WebKeys.THEME_DISPLAY);
+
+				long commerceChannelGroupId =
+					_commerceChannelLocalService.
+						getCommerceChannelGroupIdBySiteGroupId(
+							themeDisplay.getScopeGroupId());
+
+				AccountEntry accountEntry =
+					_commerceAccountHelper.getCurrentAccountEntry(
+						commerceChannelGroupId,
+						_portal.getHttpServletRequest(renderRequest));
+
+				if (accountEntry.getAccountEntryId() !=
+						commerceOrder.getCommerceAccountId()) {
+
+					return "/pending_commerce_orders/view.jsp";
```

Comment:
This should be a redirect to the pending orders page, instead of showing a different JSP.